open_spiel/python/examples/example.py

```diff
@@ -81,10 +81,11 @@ def main(_):
 
     elif state.is_simultaneous_node():
       # Simultaneous node: sample actions for all players.
-      chosen_actions = [
-          random.choice(state.legal_actions(pid))
-          for pid in range(game.num_players())
-      ]
+      def get_random_action(player):
+        legal_actions_player = state.legal_actions(player)
+        return (0 if not legal_actions_player else
+                random.choice(legal_actions_player))
+      chosen_actions = map(get_random_action, range(game.num_players())
```

Comment:
The original version (list comprehension) is preferred to `map` in Python.
The blog you linked is about map vs for loop which is a different thing. List comprehensions are (almost) always a better choice than map.

You can just leave the original code and define:
```
random_choice = lambda a: np.random.choice(a) if a else [0]
```

Reply:
Resolve comment by the last 2 commits.